### PR TITLE
Don't show game join menu item when game mode is ladder

### LINKED ISSUE #936: Ladder 1v1 games are shown in friend joined games toasts
Check for the featured mod of the game not being gw or ladder....

### LINKED ISSUE #1761: Can join ladder games through player context menu
**Describe the bug**
When a player is starting a ladder game and they are the host, they will show up in the chat tab as “hosting” and players (who should not be in the game) can join their game through the context menu causing a bugged ladder game (joining player shows up as nomads faction and is set to observer)

**To Reproduce**
(Easier on test)
1. Queue ladder with 2 players
2. Close client for the guest before the game starts.
3. Join the game with a third client through the context menu.

**Expected behavior**
Show ladder game hosts as “playing ladder” and don’t distinguish between hosts and guests.

**OS**
Linux or Windows

**Wanna have the bug fixed quickly?**
Visit Issue hunt...
[Issue hunt](https://issuehunt.io/r/FAForever/downlords-faf-client)

### PR DESCRIPTION
**Pull request by Askaholic**
Kindof hard to test on production since it's a very rare edge case. Hopefully the tests I provided are sufficient.

Fixes #1761 
Fixes #936

### CODE REVIEW

File: src/main/java/com/faforever/client/chat/ChatUserContextMenuController.java [hunk 1]
```diff
@@ -201,9 +202,11 @@ public void setChatUser(ChatChannelUser chatUser) {
       addFoeItem.visibleProperty().bind(newValue.socialStatusProperty().isNotEqualTo(FOE).and(newValue.socialStatusProperty().isNotEqualTo(SELF)));
       removeFoeItem.visibleProperty().bind(newValue.socialStatusProperty().isEqualTo(FOE));
 
+      // TODO: Make this ignore TMM games too and not just ladder
```
Comment (1-alex98) [inline src/main/java/com/faforever/client/chat/ChatUserContextMenuController.java:197]: maybe link github issue

**Review by 1-alex98 — DISMISSED** [review]

**Review by 1-alex98 — DISMISSED** [review]

**Comment by 1-alex98** [comment]
need to test it still

**Comment by Askaholic** [comment]
It’s actually not too hard to test on the test server. I did it a few times. Need 3 clients though, 2 to search ladder and one to have added the other two as a friend.

**Comment by 1-alex98** [comment]
So u tested it?

**Comment by Askaholic** [comment]
Yes

**Comment by Brutus5000** [comment]
Do we need to exclude tutorials as well?

**Comment by Askaholic** [comment]
No, tutorials aren't even registered on the server, they aren't real games.

**Comment by 1-alex98** [comment]
tested

**Comment by Askaholic** [comment]
So can we merge?

**Comment by 1-alex98** [comment]
Yes sure just forgot it
